File: java/src/test/java/io/binac/leetcode/NextPermutationTest.java
package io.binac.leetcode;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class NextPermutationTest {
    private static final NextPermutation.Solution1 SOLUTION1 = new NextPermutation.Solution1();

    private final int[] NUMS1 = {1, 2, 3};
    private final int[] EXPECTED1 = {1, 3, 2};

    private final int[] NUMS2 = {3, 2, 1};
    private final int[] EXPECTED2 = {1, 2, 3};

    private final int[] NUMS3 = {1, 1, 5};
    private final int[] EXPECTED3 = {1, 5, 1};

    private final int[] NUMS4 = {1};
    private final int[] EXPECTED4 = {1};

    @Test
    void test1() {
        SOLUTION1.nextPermutation(NUMS1);
        assertArrayEquals(EXPECTED1, NUMS1);

        SOLUTION1.nextPermutation(NUMS2);
        assertArrayEquals(EXPECTED2, NUMS2);

        SOLUTION1.nextPermutation(NUMS3);
        assertArrayEquals(EXPECTED3, NUMS3);

        SOLUTION1.nextPermutation(NUMS4);
        assertArrayEquals(EXPECTED4, NUMS4);
    }
}

File: cpp/unittests/NextPermutationTest.cpp
#include "NextPermutation.h"
#include "gtest/gtest.h"

using namespace lcpp;

TEST(NextPermutation, Solution31_1) {
  std::vector<int> Nums1{1, 2, 3};
  const std::vector<int> Expected1{1, 3, 2};
  Solution31_1().nextPermutation(Nums1);
  EXPECT_EQ(Expected1, Nums1);

  std::vector<int> Nums2{3, 2, 1};
  const std::vector<int> Expected2{1, 2, 3};
  Solution31_1().nextPermutation(Nums2);
  EXPECT_EQ(Expected2, Nums2);

  std::vector<int> Nums3{1, 1, 5};
  const std::vector<int> Expected3{1, 5, 1};
  Solution31_1().nextPermutation(Nums3);
  EXPECT_EQ(Expected3, Nums3);
}